pyqtorch/core/utils.py

```diff
@@ -86,23 +85,20 @@ def _apply_gate(
     - state (torch.Tensor): the quantum state after application of the gate.
     Same shape as `ìnput_state`
     """
-    mat = mat.reshape([2] * len(qubits) * 2)
+    mat = mat.view([2] * len(qubits) * 2)
```

Comment:
Nice this prevents a copy?

Reply:
more like, when reshape would have copied view will fail IIRC

Reply:
yes, mentioned it already in the other thread. probably it didnt copy before but like this we ensure it